### PR TITLE
[PWA-2063] Add reCaptcha to the Forgot Password Forms

### PR DESCRIPTION
**Pull request by mikhaelbois**
## Description

Create hook that can be added to forms and enable the use of Google reCaptcha V3.
First form will be the Forgot Password Form.

## Related Issue

Closes https://jira.corp.magento.com/browse/PWA-2063.

### Specification

For this to work as needed, we need the following PR to be merged in our Magento PWA module.
This enables the config validation for both Forgot Password forms.
PR: https://github.com/magento-commerce/magento2-pwa/pull/12

#### Test scenario(s) for direct fix/feature

- Enable Google reCAPTCHA v3 Invisible in the admin.
- Set `Enable for Forgot Password` to v3 Invisible in the admin.
- In the frontend, test all occurrences of Forgot Password forms (Mega menu, sidebar navigation, dedicated page and checkout) and also the Reset Password Form that you receive via email.
- Run cypress tests and validate new one `tests/integrationTests/googleReCaptcha/verifyGoogleReCaptcha.spec.js`

#### Test scenario(s) for any existing impacted features/areas

- Test different configuration scenarios to make sure the Google API is not loaded if it is not assigned to the Forgot Password form or if it is disabled.
- Make sure other forms are not impacted.
- There should only one Google API script loaded and only when needed.

#### Is Browser/Device testing needed?

- Make sure API does not impact user experience on different browsers and on mobile / desktop.

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Review by jcalcaben — CHANGES_REQUESTED** [review]
just one minor comment. other than that, it looks good.

File: packages/peregrine/lib/hooks/useScript.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { useState, useEffect } from 'react';
+
+/**
+ * Dynamically load an external script and know when its loaded
+ * @see https://usehooks.com/useScript/.
+ *
+ * @param {String} src - the script src.
+ */
```
[the remark this answers is not in the document]
Comment (mikhaelbois) [reply packages/peregrine/lib/hooks/useScript.js:10]: @jcalcaben Good point. Done!

**Review by jcalcaben — DISMISSED** [review]

**Comment by dpatil-magento** [comment]
@mikhaelbois Can get success response on graphql call and also see console error related security policy. We might need add google.com whitelist here https://github.com/magento/pwa-studio/blob/e488838019e6c77e6fd5f54f0a219699d8d4859d/packages/extensions/upward-security-headers/upward.yml#L22

image

**Comment by mikhaelbois** [comment]
@magento give me 2.4.3 instance

**Comment by mikhaelbois** [comment]
@dpatil-magento 
I've updated the CSP to include *.google frames.

**Comment by dpatil-magento** [comment]
@mikhaelbois 
1. If I hit Cancel and go back to forgot password form don't see captcha again.

<a href="https://gyazo.com/3541322e2b527c8e87a4734fa9fb2168">Image from Gyazo</a>

And if I try to submit email in above case, I get captcha failed error.

2. When configured to display Bottom right , The captcha does not go away until we reload page. I was expecting it to go away after we submit email or if I navigate to some new page using mega menu.

image

**Comment by mikhaelbois** [comment]
@dpatil-magento 
1. I'll resolve this issue.
2. Since the script is loaded once and stays active until we reload the page, I don't think we should or can hide the badge. Also, with the floating menu, it might become confusing to see the badge come and go every time the menu is toggled.

**Review by ericeoeur — APPROVED** [review]
LGTM. QA has passed for this ticket.

**Pull request merged**